test/fact_importer_test_helper.rb

```diff
@@ -34,3 +34,119 @@ def instance_stubs
     @instance_stubs ||= []
   end
 end
+
+module FactsData
+  class FlatFacts
+    def filter
+      ['ignore*', '*_bad', 'filter']
+    end
+
+    def good_facts
+      {
+        'just_a_fact' => 'hello',
+        'nofilter' => 'hello'
+      }
+    end
+
+    def ignored_facts
+      {
+        'ignored_fact' => 'will_not_show',
+        'fact_bad' => 'will_not_show',
+        'filter' => 'will_not_show',
+        'ignore' => 'will_not_show',
+        '_bad' => 'will_not_show'
+      }
+    end
+  end
+
+  class RhsmStyleFacts
+    def filter
+      ['ignore*', '*_bad', 'filter']
+    end
+
+    def good_facts
+      {
+        'something::something_else' => 'hello'
+      }
+    end
+
+    def ignored_facts
+      {
+        'ignored_fact::something' => 'will_not_show',
+        'something::ignored_fact' => 'will_not_show',
+        'something::ignored_fact::something_else' => 'will_not_show',
+        'fact_bad::something' => 'will_not_show',
+        'something::fact_bad' => 'will_not_show',
+        'something::fact_bad::something_else' => 'will_not_show',
+        'filter::something' => 'will_not_show',
+        'something::filter' => 'will_not_show',
+        'something::filter::something_else' => 'will_not_show'
+      }
+    end
+  end
```

Comment:
One could say this belongs to Katello, but I really think it should be in core. Thanks.

Reply:
Katello is only an example of this kind of facts. Since this notation is supported by core, I think it should be tested in core.